client/send-video.cc

```diff
@@ -118,148 +132,167 @@ int main(int argc, char *argv[]) {
+                                    pCodecCtx->height);
+        buffer=(uint8_t *)av_malloc(numBytes*sizeof(uint8_t));
+
+        // Assign appropriate parts of buffer to image planes in pFrameRGB
+        // Note that pFrameRGB is an AVFrame, but AVFrame is a superset
+        // of AVPicture
+        avpicture_fill((AVPicture *)pFrameRGB, buffer, AV_PIX_FMT_RGB24,
+                       pCodecCtx->width, pCodecCtx->height);
+
+        // initialize SWS context for software scaling
+        sws_ctx = sws_getContext(pCodecCtx->width,
+                                 pCodecCtx->height,
+                                 pCodecCtx->pix_fmt,
+                                 display.width(), display.height(),
+                                 AV_PIX_FMT_RGB24,
+                                 SWS_BILINEAR,
+                                 NULL,
+                                 NULL,
+                                 NULL
+                                 );
+        if (sws_ctx == 0) {
+            fprintf(stderr, "Trouble doing scaling to %dx%d :(\n",
+                    display.width(), display.height());
+            return 1;
+        }
 
-    if (off_z > 0) {
-        display.Clear();
-        display.Send();
-    }
+        signal(SIGTERM, InterruptHandler);
+        signal(SIGINT, InterruptHandler);
+
+        // Read frames and send to FlaschenTaschen.
+        const int frame_wait_micros = 1e6 / fps;
+        const tmillis_t startPlay = GetTimeInMillis(); //mark time
+        int repeatedcount = 0;
+        do {
+          frame_count = 0;
+          while (!interrupt_received && av_read_frame(pFormatCtx, &packet) >= 0) {
+              // Is this a packet from the video stream?
+              if (packet.stream_index==videoStream) {
+                  // Decode video frame
+                  avcodec_decode_video2(pCodecCtx, pFrame, &frameFinished, &packet);
+
+                  // Did we get a video frame?
+                  if (frameFinished) {
+                      // Convert the image from its native format to RGB
+                      sws_scale(sws_ctx, (uint8_t const * const *)pFrame->data,
+                                pFrame->linesize, 0, pCodecCtx->height,
+                                pFrameRGB->data, pFrameRGB->linesize);
+
+                      // Save the frame to disk
+                      SendFrame(pFrameRGB, &display);
+                      frame_count++;
+                  }
+                  usleep(frame_wait_micros);
+              }
+
+              // Free the packet that was allocated by av_read_frame
+              av_free_packet(&packet);
+          }
+          if ((GetTimeInMillis() - startPlay) < (repeatSeconds * 1000) && !av_seek_frame(pFormatCtx, -1, 1, AVSEEK_FLAG_FRAME)) {
+              if (repeatedcount)
+                  fprintf(stderr, "finished loop %d (%ld frames) after %0.1fs\n", repeatedcount, frame_count, (GetTimeInMillis() - startPlay) / 1000.);
```

Comment:
This repeatedcount looks like some leftover debugging printf while developing this feature, but do you think it will be useful for the user in the end ? I think it just is somewhat confusing without much merit for some enduser.
I'd probably just remove it for simplicity reasons.